README.md

```diff
@@ -113,17 +114,37 @@ julia> residual!(model, model.meta.x0, model.F)
  -0.4486499211288866
 ```
 
-You need to call ``jac_structure_residual!`` at least once before calling ``jac_op_residual!``.
+```julia
+julia> meta_nls = nls_meta(model)
+julia> S = typeof(model.meta.x0)
+julia> F = S(undef, meta_nls.nequ)
+julia> residual!(model, model.meta.x0, F)
+63686-element Vector{Float64}:
+ -9.020226301243213
+ 11.263958304987227
+  ⋮
+ -0.01443314653496941
+ -0.4486499211288866
+```
+
+You need to call `jac_structure_residual!` at least once before calling `jac_op_residual`.
 
 ```julia
-julia> jac_structure_residual!(model, model.rows, model.cols)
-([1, 1  …  63686, 63686], [1, 2  …  23768, 23769])
+julia> S = typeof(model.meta.x0)
+julia> meta_nls = nls_meta(model)
+julia> rows = Vector{Int}(undef, meta_nls.nnzj)
+julia> cols = Vector{Int}(undef, meta_nls.nnzj)
+julia> jac_structure_residual!(model, rows, cols)
+([1, 1 … 63686, 63686], [1, 2 … 23768, 23769])
 ```
 
-You need to call ``jac_coord_residual!`` everytime before calling ``jac_op_residual!``.
+You need to call `jac_coord_residual!` everytime before calling `jac_op_residual`.
```

Comment:
why?

Reply:
The `jac_op_residual` function doesn't update the matrix so it needs to be done manually at each iteration.